### PR TITLE
add tanhshrink op

### PR DESCRIPTION
**Pull request by beverlylytle**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Add support for the tanhshrink activation function.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: thunder/tests/opinfos.py [hunk 1]
```diff
@@ -1811,6 +1811,30 @@ def hardshrink_opinfo_factory(lambds):
 elementwise_unary_ops.append(selu_opinfo)
 
 
+tanhshrink_opinfo = OpInfo(
+    ltorch.tanhshrink,
+    dtypes=(datatypes.inexact,),
+    sample_input_generator=elementwise_unary_generator,
+    torch_reference=torch.nn.functional.tanhshrink,
+    test_directives=(
+        # Torch doesn't support CPU float16 or complex32 tanhshrink
```
Comment (mruberry) [inline thunder/tests/opinfos.py:1820]: Nice comment

File: thunder/tests/opinfos.py [hunk 2]
```diff
@@ -1811,6 +1811,30 @@ def hardshrink_opinfo_factory(lambds):
...
+        DecorateInfo(
+            pytest.mark.xfail,
+            "test_core_vs_torch_consistency",
+            dtypes=(datatypes.float16, datatypes.complex32),
+            devicetypes=(devices.DeviceType.CPU,),
+        ),
+        # NOTE Nvfuser fails with AssertionError: Tensor-likes are not close!
```
Comment (mruberry) [inline thunder/tests/opinfos.py:1827]: Interesting! How far off are the results? Let's include the atol and rtol in the comment. Can we set a custom comparator to get the tests to pass?

**Review by mruberry — COMMENTED** [review]
I just have a question about how far off the nvFuser numbers are. If they're really far off we should file an issue to track them (and add a link to that issue in a comment before the skip)

File: thunder/tests/opinfos.py (hunk 2 above)
Comment (beverlylytle) [reply thunder/tests/opinfos.py]: Sure, custom comparison added.

File: thunder/torch/__init__.py [hunk 3]
```diff
@@ -1905,6 +1905,11 @@ def silu(a: TensorLike, /, inplace: bool = False) -> TensorLike:
 _inplace_to_out_of_place[silu] = silu, 1
 
 
+@torchsymbol(torch.nn.functional.tanhshrink)
+def tanhshrink(a: TensorLike, /):
```
[the remark this answers is not in the document]
Comment (beverlylytle) [reply thunder/torch/__init__.py]: Why can I never remember this.  Thanks for being my linter.

File: thunder/tests/opinfos.py [hunk 4]
```diff
@@ -1804,6 +1804,31 @@ def hardshrink_singularity_fn_producer(sample: SampleInput):
...
+    test_directives=(
+        # Torch doesn't support CPU float16 or complex32 tanhshrink
+        DecorateInfo(
+            pytest.mark.xfail,
+            "test_core_vs_torch_consistency",
+            dtypes=(datatypes.float16, datatypes.complex32),
+            devicetypes=(devices.DeviceType.CPU,),
+        ),
+        DecorateInfo(
+            custom_comparator(partial(assert_close, atol=1e-2, rtol=1e-2)),
```
Comment (mruberry) [inline thunder/tests/opinfos.py:1821]: Nice custom comparator

Comment (mruberry) [reply thunder/tests/opinfos.py:1821]: fyi @kevinstephano -- maybe worth an nvfuser issue because these tolerances are pretty far off?

**Pull request merged**